src/app/[locale]/clientSearchDataset.ts

```diff
@@ -0,0 +1,25 @@
+import {SearchOptions, SearchResult} from '@/lib/dataset-fetcher';
+
+interface SearchDatasets {
+  licenses: string[];
+  publishers: string[];
+}
+
+// only use this function for client components
+// For server component use the dataset-fetcher directly
+export default async function clientSearchDatasets({
+  licenses,
+  publishers,
+}: SearchDatasets): Promise<SearchResult> {
+  const options: SearchOptions = {
+    filters: {publishers, licenses},
+  };
+  const response = await fetch('/api/dataset-search', {
+    method: 'POST',
+    body: JSON.stringify(options),
+  });
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
```

Comment:
Curious: is it possible to make a user-friendly (e.g. localized) version of this error message, for display in the UI?

Reply:
Yes. For now, I have some problems with localization. (https://github.com/colonial-heritage/dataset-browser/pull/46#issuecomment-1413418176)

But what I can do, is display this message in the error message for the users.